### ISSUE TITLE
[Feature Request] Support For Ghost Recon Future Soldier

### ISSUE THREAD

**Issue by slidenerd** [unixoide/5th-echelon#11]
- I am not sure if this is the right place to reach out so I apologize if I am in the wrong here
- Ghost Recon Future Soldier uses the same system as blacklist or atleast a very very highly similar system
- It was one of the best third person PvP games when I came out and Ubisoft has completely thrown it into the dustbin like Blacklist
- I request you to consider adding support for playing Future Soldier or making a fork of this repo that lets you do the same
- We have a huge community with 1000+ people on PC similar to the Blacklist discord servers and this would be an absolute game changer
- Kindly consider our request

**Comment by cloudslsw** [comment]
from same discord server bump for this, if its feasible can you do our community a favor?  i mean if its easily achieved with your spare time can you considering making one for future soldier?  our demand not mandatory but just a little hope up there upon what you done on similar backend

**Comment by unixoide** [comment]
Unfortunately, the latest fix release is not the community server (yet). But only makes the official servers usable again. So once I was able to develop an independent server, I might can take a look at future soldier. From what I can tell, there are other [projects](https://github.com/Arganom/Overlord) for this game though.

**Comment by cloudslsw** [comment]
> Unfortunately, the latest fix release is not the community server (yet). But only makes the official servers usable again. So once I was able to develop an independent server, I might can take a look at future soldier. From what I can tell, there are other [projects](https://github.com/Arganom/Overlord) for this game though.

yes we are also in same discord server with Arganom, he indeed is on coding a FS private server but this one just a game launcher he made with fixes in order to playing on official server, sorry we get it wrong on this project aimed, dang we original thought that was an private server emulator lol

**Comment by Coinhexa** [comment]
> Unfortunately, the latest fix release is not the community server (yet). But only makes the official servers usable again. So once I was able to develop an independent server, I might can take a look at future soldier. From what I can tell, there are other [projects](https://github.com/Arganom/Overlord) for this game though.

That is a launcher that fixes some issues in the game like installing punkbuster etc. It is not a private server. Assuming Arganom has already reached out to you, he has the wireshark packet capture files that would be necessary. You would be doing a huge favor as both games share the same netcode from what Arganom managed to research. Please do consider

**Comment by Pensellead** [comment]
Please please please help us with Ghost recon future soldier 😎

**Comment by unixoide** [comment]
Very generous, but I don't take donations. This is just a hobby project and should not result in any benefits besides making abandoned games working again.

If this projects works out for Blacklist, I will consider to take a look at other games who do have similar protocols. But it will take time and I can't promise anything.

**Comment by michal-kapala** [comment]
I published my unfinished rdv emulator for grfs, contributions are welcome
https://github.com/Phoenix-Network-Ltd/Ghost-Recon-Future-Soldier-Research

**Comment by slidenerd** [comment]
@unixoide mimak managed to decrypt packets with his quazal protocol inspector libraries [here](https://github.com/Phoenix-Network-Ltd/Ghost-Recon-Future-Soldier-Research) which I believe was the biggest challenge If it isnt too much to ask, could you take a look into how this can be used for matchmaking in this game

**Comment by Ston3Cold** [comment]
@unixoide First of all - thanks a lot for creating this server emulator for SC:BL. It works perfectly. I'll be perfectly honest, I'm not a programmer by a longshot, just an enthusiast with a huge passion for game restoration projects. 

Since I've discovered this, like others in this thread, I've tried my hand at modifying it for GRFS. I've mainly focused my efforts on modifying the launcher for it to detect the GRFS exe files. I've managed to get the installation directory modified to pickup my Steam installation of GRFS.

However, the launcher just doesn't want to detect the binaries for GRFS after modifying the necessary entries in the source to point to the exe's. I'm flabbergasted as to what the reason could be. The main difference in the exe's is that GRFS has 4 exe's vs the 2 exe's of SCBL.

GRFS has:
Launcher.exe
Furure Soldier.exe
Future Soldier DX9.exe
Future Soldier DX11.exe

I've changed the source to use the DX9 and DX11 exe's but no dice. What could possibly the reason as to why it doesn't detect them? I've checked through all the other project files, but don't see any references to additional entries for game exes.

If you could provide me with some pointers as to what to look for / modify, I'll really appreciate it.

Thanks
SC

**Comment by slidenerd** [comment]
just here to ask before the issue gets closed, anyone figured anything out on this?

**Comment by unixoide** [comment]
No, haven't put any work into GRFS so far (also because I don't own the game and never played it). But @michal-kapala project looks quite promising?

**Comment by michal-kapala** [comment]
I practically haven't developed it since as im more focused on equivalent projects for Assassin's Creed titles, however there's a fresh early-days attempt at a GRFS emulator - https://github.com/koteykaby/raven
